x-pack/filebeat/processors/aws_vpcflow/internal/strings/strings.go

```diff
@@ -60,25 +70,29 @@ func Fields(s string) []string {
 		fieldStart = i
 	}
 	if fieldStart < len(s) { // Last field might end at EOF.
-		a[na] = s[fieldStart:]
+		if na >= len(dst) {
+			return na, errTooManySubstrings
+		}
+		dst[na] = s[fieldStart:]
+		na++
 	}
-	return a
+	return na, nil
 }
 
-// FieldsFunc splits the string s at each run of Unicode code points c satisfying f(c)
+// fieldsFunc splits the string s at each run of Unicode code points c satisfying f(c)
 // and returns an array of slices of s. If all code points in s satisfy f(c) or the
 // string is empty, an empty slice is returned.
 //
 // FieldsFunc makes no guarantees about the order in which it calls f(c)
 // and assumes that f always returns the same value for a given c.
-func FieldsFunc(s string, f func(rune) bool) []string {
+func fieldsFunc(dst []string, s string, f func(rune) bool) (int, error) {
 	// A span is used to record a slice of s of the form s[start:end].
 	// The start index is inclusive and the end index is exclusive.
 	type span struct {
 		start int
 		end   int
 	}
-	spans := make([]span, 0, 32)
+	spans := make([]span, 0, len(dst))
```

Comment:
This is unfortunate. Are will still winning compared with a call to stdlib `strings.Fields` and then assesing the length of the returned slice?